lib/chef/provider/package/windows/exe.rb

```diff
@@ -0,0 +1,177 @@
+          end
+
+          # Returns a version if the package is installed or nil if it is not.
+          def installed_version
+            Chef::Log.debug("#{@new_resource} checking package version")
+            current_installed_version
+          end
+
+          def package_version
+            @new_resource.version || install_file_version
+          end
+
+          def install_package
+            Chef::Log.debug("#{@new_resource} installing #{@new_resource.installer_type} package '#{@new_resource.source}'")
+            shell_out!(
+              [
+                "start",
+                "\"\"",
+                "/wait",
+                "\"#{@new_resource.source}\"",
+                unattended_flags,
+                expand_options(@new_resource.options),
+                "& exit %%%%ERRORLEVEL%%%%"
+              ].join(" "), timeout: @new_resource.timeout, returns: @new_resource.returns
+            )
+          end
+
+          def remove_package
+            if @new_resource.version.nil? && installed_packages.count > 1
+              raise Chef::Exceptions::MultiplePackagesFound, "Removing Windows Package '#{@new_resource.name}' found versions '#{@new_resource.version}' and no version given"
+            end
+
+            uninstall_version = @new_resource.version || installed_packages.keys[0]
+            uninstall_string = installed_packages[uninstall_version][:uninstall_string]
+
+            Chef::Log.debug("Registry provided uninstall string for #{@new_resource} is '#{uninstall_string}'")
+            shell_out!(uninstall_command(uninstall_string), { returns: @new_resource.returns })
+          end
+
+          private
+
+          def uninstall_command(uninstall_string)
+            uninstall_string.delete!('"')
+            if uninstall_string.start_with?("MsiExec.exe ")
+              uninstall_string += expand_options(@new_resource.options)
+              uninstall_string += " /Q" unless uninstall_string =~ / \/Q\b/
+            else
+              uninstall_string = [
+                %q{/d"},
+                ::File.dirname(uninstall_string),
+                %q{" },
+                ::File.basename(uninstall_string),
+                expand_options(@new_resource.options),
+                " ",
+                unattended_flags
+              ].join
+            end
+            %Q{start "" /wait #{uninstall_string} & exit %%%%ERRORLEVEL%%%%}
+          end
+
+          def current_installed_version
+            installed_packages.count == 0 ? nil : installed_packages.keys
+          end
+
+          def install_file_version
+            @install_file_version ||= begin
+              if @new_resource.source.nil?
+                nil
+              else
+                version_info = Chef::ReservedNames::Win32::File.version_info(@new_resource.source)
+                file_version = version_info.FileVersion || version_info.ProductVersion
+                file_version == '' ? nil : file_version
```

Comment:
`file_version.empty? ? nil : file_version` ?

the double question marks look wierd, but ` == ''` looks weird to me. *shrug*